### PR TITLE
Radical Solana refresh optimizations

### LINKED ISSUE #15641: User cannot send Solana because we set low fees
**Describe the bug**
Users are sometimes not able to send Solana transactions and have to try repeatedly. 
This is most most likely happens when fees on the network are high, because we are setting a very low hardcoded minimum fee

User report: https://www.reddit.com/r/TREZOR/comments/1h138m4/comment/lzazrln/

> Sometimes when i try to send my USDC to another wallet i encounter this
> error:
> TransactionExpiredBlockheightExceededError: Signature
> sometimes I have to wait like 20 minutes and try to send again later in hopes it will work or spam it a lot until it goes through. I also read it could be because of network congestion and trezor putting the lowest fees possible
> 

**How to reproduce**

1. Go to Solana account, when the average network fees are high
2. Try to send some tokens
3. The transaction might fail because of fees and you get: `TransactionExpiredBlockheightExceededError: Signature`

**Expected behavior**
The transaction does not fail, is approved, and user is able to send tokens on first try

### PR DESCRIPTION
**Pull request by martykan**
## Description

Reduce the amount of periodic calls for Solana

1. Check if the account was updated only based on txids. This means that it uses only 1 request per account + 1 per token. Previously this was more than 5 requests per account + 1 per token. 
2. Remove Solana block subscription from Suite's side and move it to send form only, at a reduced interval of 50 seconds.
3. Fixed baseFee calculation, increase priority fee unit price x3, added simulations for precise calculation of compute limit, resulting in overall lower fees but higher priority. 
4. Fixed Solana account subscriptions. Added solution to reconnect Solana backend when subscription goes down.
5. Changed configuration for sending TX to run the preflight check and allow retries
6. Fixed "send max" in send form

## Benchmark

Count of all Solana requests when opening a wallet with 4 accounts and several tokens and waiting for 5 min (including discovery)

- Before: 449 (121 discovery + 82 per minute)
- After: 200 (72 discovery + 32 per minute)

### CODE REVIEW

File: packages/blockchain-link-types/src/params.ts [hunk 1]
```diff
@@ -56,4 +56,5 @@ export interface AccountInfoParams {
         ledger: number;
         seq: number;
     };
+    tokenAccounts?: string[]; // solana only, token accounts to fetch txids for
```
Comment (vytick) [inline packages/blockchain-link-types/src/params.ts]: wdyt about smthng like
```suggestion
    tokenAccountsPubKeys?: string[]; // solana only, token accounts to fetch txids for
```

File: packages/blockchain-link/src/workers/solana/fee.ts [hunk 2]
```diff
@@ -35,31 +54,40 @@ export const getBaseFee = async (api: Connection, message: Message) => {
...
+        .times(computeUnitLimit.toString())
         .decimalPlaces(0, BigNumber.ROUND_UP)
-        .toNumber();
+        .toString(10);
 
-    return { computeUnitPrice: computeUnitPrice.toNumber(), computeUnitLimit, fee };
+    return {
+        computeUnitPrice,
+        computeUnitLimit,
+        fee: BigInt(fee) as MicroLamports,
```
Comment (vytick) [inline packages/blockchain-link/src/workers/solana/fee.ts]: It seems that here it is casting to microlamports but previous computation of fee already did microlamports -> lamports 🤔. Maybe this is not needed at all or am I wrong?

Comment (martykan) [reply packages/blockchain-link/src/workers/solana/fee.ts]: Yeah seems wrong and also it doesn't really do anything, but this should be dealt with in [solana-web3-js-two-dot-oh](https://github.com/trezor/trezor-suite/pull/15536). 
I will change the target branch here to separate the commits

**Review by vytick — APPROVED** [review]
Looks good 👍

**Review by vytick — APPROVED** [review]

**Review by tomasklim — APPROVED** [review]

**Comment by bosomt** [comment]
**QA OK**

Solana works as expected, 
- discovery, add accounts
- transactions - sol `0.23541798 RENDER sent from Solana #2` and tokens `0.23541798 RENDER sent from Solana #2`


**Info:**
 - Suite version: desktop 24.12.0 (f7dd25b803538b443f07f880f4fdceaa4faadd1a)
 - Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) TrezorSuiteDev/24.12.0 Chrome/130.0.6723.91 Electron/33.1.0 Safari/537.36
 - OS: MacIntel
 - Screen: 1512x982
 - Device: Trezor T3T1 2.8.6 regular (revision 688bddc887b8bb660640cc6e40a99036f2e87f1f)
 - Transport: BridgeTransport 2.0.33

**Pull request merged**